### PR TITLE
Azure - Add handling of Linux and Consumption App Service Plans

### PR DESCRIPTION
**Pull request by aluong**
- Linux plans need the `reserved` keyword set when updating the resource
- Consumption or Function plans should not be allowed to be resized
- For `provision.sh`, I am changing the deployment mode to `complete` to destroy non-described resoruces
- For `cleanup.sh`, Azure Policy cannot be done async.

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/resources/appserviceplan.py [hunk 1]
```diff
@@ -90,6 +90,14 @@ def _prepare_processing(self):
     def _process_resource(self, resource):
         model = models.AppServicePlan(location=resource['location'])
 
+        if resource['kind'] == 'functionapp':
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/appserviceplan.py]: Should we just get rid of consumption plans when we get a list of all app service plan? (There is nothing we can configure\control, right?)

File: tools/c7n_azure/tests/test_app_service_plan.py [hunk 2]
```diff
@@ -46,71 +48,108 @@ def test_app_service_plan_schema_validate(self):
             self.assertTrue(p)
 
     @arm_template('appserviceplan.json')
-    def test_find_by_name(self):
+    def test_resize_plan_win(self):
```
Comment (logachev) [inline tools/c7n_azure/tests/test_app_service_plan.py]: You won't be able to run this test twice, right?

File: tools/c7n_azure/c7n_azure/resources/appserviceplan.py (hunk 1 above)
Comment (aluong) [reply tools/c7n_azure/c7n_azure/resources/appserviceplan.py]: We can actually still perform other generic actions like tag

File: tools/c7n_azure/tests/test_app_service_plan.py (hunk 2 above)
Comment (aluong) [reply tools/c7n_azure/tests/test_app_service_plan.py]: Correct. You would need to re-provision or set it up again.

Comment (stefangordon) [reply tools/c7n_azure/tests/test_app_service_plan.py]: That isn't ideal   ;)   Took forever to fix all those last time.   I'd probably make/edit/fix resource in the setup/teardown (like the old test-storage-firewall-actions does).  Or just patch out the actual HTTP calls.

**Comment by aluong** [comment]
fix #4556

**Pull request merged**